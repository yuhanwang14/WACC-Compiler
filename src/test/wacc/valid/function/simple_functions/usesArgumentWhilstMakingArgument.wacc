# tests a function with more arguments than fit into registers

# Output:
# 12
# -4
# 32
#

# Program:

begin
  int f(int u, int v) is
    int x = call g(u + v, u - v, u * v) ;
    return x
  end

  int g(int x, int y, int z) is
    println(x) ;
    println(y) ;
    println(z) ;
    return 0
  end

  int r = call f(4, 8)
end
