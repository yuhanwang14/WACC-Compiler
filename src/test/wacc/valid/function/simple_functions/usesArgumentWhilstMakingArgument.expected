12
-4
32

